### PR TITLE
Add arraylake client

### PR DESCRIPTION
**Pull request by jhamman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
~- [ ] If static libraries are linked in, the license of the static library is packaged.~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Questions for maintainers: This package is intentionally unlicenced for the moment, and while that will change in the future, we would still like to get this wired together. Is this something Conda-forge can support or is an OSS license required?

### CODE REVIEW

**Comment by jhamman** [comment]
@conda-forge/help-python - I could use some help skipping windows in this build. I thought `noarch` would do this but I guess not.

File: recipes/arraylake-client/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/arraylake_client-{{ version }}.tar.gz
+  sha256: e01b97f779089fde2ec3aae58e32034be8c81a93caec5e4c62a5ac0b778a68fa
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (ocefpaf) [inline recipes/arraylake-client/meta.yaml:15]: This should fix the Windows build

```suggestion
  number: 0
  entry_points:
    - arraylake = arraylake_client.cli.main:app
```

File: recipes/arraylake-client/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - zarr >=2.14,<2.15
+    - numpy >=1.20.0,<2.0.0
+    - donfig >=0.7,<1.0
+    - pydantic >=1.10.0,<2.0.0
+    - httpx >=0.23.0,<1.0
+    - pymongo >=4.2.0,<5.0.0
+    - cachetools >=5.3.0,<6.0.0
+    - structlog >=23.1.0,<24.0.0
+    - email_validator
+    - uvloop  # [not win]
```
Comment (ocefpaf) [inline recipes/arraylake-client/meta.yaml:37]: If this is optional you can make it noarch, and available on Windows ,by adding both OS-noarch selectors. If this is not optional you can still make it noarch but you cannot add the Windows option. See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies for more info.

```suggestion
    - uvloop  # [not win]
```

**Comment by ocefpaf** [comment]
There is a bug in the linter when using the os-noarch, we can ignore it for now.

**Comment by jhamman** [comment]
Thanks for the help @ocefpaf!

**Pull request merged**